### ISSUE TITLE
MissingConstructor issue emitted erroneously with trait constructor

### ISSUE THREAD

**Issue by muglug** [vimeo/psalm#173]
```php
trait T {
  /** @var string **/
  public $foo;
  
  public function __construct() {
    $this->foo = "hello";
  }
}

class A {
  use T;
}
```

This code should be fine

**Comment by oruborus** [comment]
Hello and thank you for your work!

There might be another false positive in this regard.

```php
trait T {
  /** @var string **/
  public $foo;
  
  public function bar() {
    $this->foo = "hello";
  }
}

class A {
  use T {
    bar as __construct;
  }
}
```
